### PR TITLE
Qu auto/sorqa 84 Contacts filters for DE

### PR DESCRIPTION
**Pull request by mk-sgent**

### CODE REVIEW

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CreateNewCaseSteps.java [hunk 1]
```diff
@@ -181,6 +192,30 @@ public CreateNewCaseSteps(WebDriverHelpers webDriverHelpers, CaseService caseSer
...
+          selectResponsibleRegion(caze.getResponsibleRegion());
+          selectResponsibleDistrict(caze.getResponsibleDistrict());
+          selectResponsibleCommunity(caze.getResponsibleCommunity());
+
+          selectPlaceOfStay(caze.getPlaceOfStay());
+          fillPlaceDescription(caze.getPlaceDescription());
+          selectPresentConditionOfPerson(caze.getPresentConditionOfPerson());
+          fillDateOfSymptomOnset(caze.getDateOfSymptomOnset(), Locale.GERMAN);
+          webDriverHelpers.selectFromCombobox(
+              CASE_DISEASE_VARIANT_COMBOBOX, "Sequenzierung ausstehend");
```
Comment (jp-sgent) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CreateNewCaseSteps.java:212]: If there are multiple variants that can be chosen from the combobox, it might me good to vary the data that test inputs.

File: sormas-e2e-tests/src/test/resources/features/sanity/web/ContactFilters.feature [hunk 2]
```diff
@@ -115,4 +115,108 @@ Feature: Contact filter functionality
...
+
+  @issue=SORQA-5911 @env_de
+  Scenario: Check all filters are working properly in Contacts directory
+    Given I log in with National User
+    When I click on the Contacts button from navbar
+    And I click on the NEW CONTACT button
+    And I fill a new contact form for DE version
+    And I click SAVE button on Edit Contact Page
+    Then I check the created data is correctly displayed on Edit Contact page for DE version
+    And I click on BESTÄTIGTER KONTAKT radio button Contact Person tab
```
Comment (jp-sgent) [inline sormas-e2e-tests/src/test/resources/features/sanity/web/ContactFilters.feature:181]: It it generally not advised to use a different language in the step definition. It's better to write something like "I click on the German button for Confirmed Contact", so that someone who doesn't speak German can make sense of what the scenario is doing. Same applies to all the other mixed-language steps below.

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CreateNewCaseSteps.java [hunk 3]
```diff
@@ -181,6 +192,30 @@ public CreateNewCaseSteps(WebDriverHelpers webDriverHelpers, CaseService caseSer
...
+        "^I create a new case for contact with specific data for DE$",
+        () -> {
+          caze = caseService.buildGeneratedCaseDE();
+          fillDateOfReport(caze.getDateOfReport(), Locale.GERMAN);
+          selectCaseOrigin(caze.getCaseOrigin());
+          fillExternalId(caze.getExternalId());
+          selectResponsibleRegion(caze.getResponsibleRegion());
+          selectResponsibleDistrict(caze.getResponsibleDistrict());
+          selectResponsibleCommunity(caze.getResponsibleCommunity());
+
```
Comment (mp-sgent) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CreateNewCaseSteps.java:206]: remove space

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java [hunk 4]
```diff
@@ -117,6 +134,34 @@ public EditContactSteps(
...
+                  "caseOrEventInformation",
+                  "responsibleRegion",
+                  "responsibleDistrict",
+                  "responsibleCommunity",
+                  "additionalInformationOnContactType",
+                  "typeOfContact",
+                  "contactCategory",
+                  "relationshipWithCase",
+                  "descriptionOfHowContactTookPlace"));
+          TimeUnit.SECONDS.sleep(6);
```
Comment (mp-sgent) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java:162]: why sleep?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java [hunk 5]
```diff
@@ -44,14 +50,14 @@
...
 public class EditContactSteps implements En {
+  DateTimeFormatter formatter = DateTimeFormatter.ofPattern("M/d/yyyy");
   private final WebDriverHelpers webDriverHelpers;
   public static Contact createdContact;
   public static Contact collectedContact;
   public static QuarantineOrder aQuarantineOrder;
   public static Contact editedContact;
   public static Contact aContact;
   public static final String userDirPath = System.getProperty("user.dir");
-  public static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("M/d/yyyy");
```
Comment (mp-sgent) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java:54]: why did you remove it and instead addded code in 53 line?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java [hunk 6]
```diff
@@ -209,9 +213,17 @@ public ContactDirectorySteps(
...
+          webDriverHelpers.fillAndSubmitInWebElement(
+              CONTACT_DIRECTORY_DETAILED_PAGE_FILTER_INPUT, contactUuid);
+        });
+
+    When(
+        "I apply Id of last created Contact on Contact Directory Page",
         () -> {
           String contactUuid =
-                  apiState.getCreatedContact().getUuid();
+              dataOperations.getPartialUuidFromAssociatedLink(collectedContact.getUuid());
```
Comment (mp-sgent) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ContactDirectorySteps.java:226]: better would be to fulfill filter field with whole UUID not partial, can cause problems with mutliple records in a grid

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java [hunk 7]
```diff
@@ -90,6 +96,17 @@ public EditContactSteps(
                   "descriptionOfHowContactTookPlace"));
         });
 
+    When(
```
Comment (mp-sgent) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java:99]: This method can be removed, instead collection of UUID can be implemented in collectContactDataDE() with rest of Contact attributes on Edit Contact Page

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java [hunk 8]
```diff
@@ -718,4 +763,20 @@ private Contact collectContactPersonUuid() {
     webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(UUID_INPUT, 40);
     return Contact.builder().uuid(webDriverHelpers.getValueFromWebElement(UUID_INPUT)).build();
   }
+
+  private Contact collectContactUuid() {
+    return Contact.builder().uuid(webDriverHelpers.getValueFromWebElement(UUID_INPUT)).build();
```
Comment (mp-sgent) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java:768]: can be moved to collectContactDataDE()

File: sormas-e2e-tests/src/test/resources/features/sanity/web/ContactFilters.feature (hunk 2 above)
Comment (mp-sgent) [reply sormas-e2e-tests/src/test/resources/features/sanity/web/ContactFilters.feature:181]: True, but in this case we are sharing this method across instances cause it enables clicking web element by text (BESTÄTIGTER KONTAKT is a parameter that you can define in this test step)

**Comment by mk-sgent** [comment]
**First**
https://jenkins.sormas.netzlink.com/job/sormas-RunE2eTests-branch/292/allure/

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CreateNewCaseSteps.java [hunk 9]
```diff
@@ -257,6 +260,12 @@ public CreateNewCaseSteps(
           webDriverHelpers.waitForPageLoadingSpinnerToDisappear(40);
         });
 
+    When(
+        "I search for the last case uuid in the CHOOSE SOURCE window for UI",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CreateNewCaseSteps.java:264]: not for UI, from which view is this action

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CreateNewCaseSteps.java [hunk 10]
```diff
@@ -257,6 +260,12 @@ public CreateNewCaseSteps(
...
+        () -> {
+          webDriverHelpers.fillInWebElement(SOURCE_CASE_WINDOW_CASE_INPUT, aCase.getUuid());
+          webDriverHelpers.clickOnWebElementBySelector(SOURCE_CASE_WINDOW_SEARCH_CASE_BUTTON);
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CreateNewCaseSteps.java:267]: add after a wait for locator (uuid locator formated based on provided value)

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java [hunk 11]
```diff
@@ -64,9 +70,16 @@ public EditContactSteps(
     this.webDriverHelpers = webDriverHelpers;
 
     When(
-        "I check the created data for DE version is correctly displayed on Edit Contact page",
+        "I open the last created contact by UI",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java:73]: not by UI, from which directory/view was created

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java [hunk 12]
```diff
@@ -492,6 +505,8 @@ private boolean isFollowUpNotVisible() {
   private Contact collectContactDataDE() {
     String collectedDateOfReport = webDriverHelpers.getValueFromWebElement(REPORT_DATE);
     String collectedLastDateOfContact = webDriverHelpers.getValueFromWebElement(LAST_CONTACT_DATE);
+    //    collectedContact =
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java:508]: why commented?

File: sormas-e2e-tests/src/test/resources/features/sanity/web/ContactFilters.feature (hunk 2 above)
Comment (rdutu-vg) [reply sormas-e2e-tests/src/test/resources/features/sanity/web/ContactFilters.feature:174]: agree with @jp-sgent , is not ok to use any other language in the BDD layer excepting english, because when we'll check the steps, we won't be able to understand what is there.
Please use the main one, and adapt the locator so it won't be taken by ui text.

**Review by rdutu-vg — CHANGES_REQUESTED** [review]

**Review by rdutu-vg — APPROVED** [review]
